execute_function_1_when_function_2_returns_true.js

```diff
@@ -1,25 +1,10 @@
-<script type="text/javascript">
-
-	function execute_function_1_when_function_2_returns_true( first_function_parementer, second_function_parementer, check_frequency_parameter ){
-
-		if( typeof check_frequency_parameter == 'undefined' ) var check_frequency_parameter = 100;
-	
-		if( second_function_parementer() == true ) {
-			
-			first_function_parementer();
-			
-		}
-		
-		else {
-			
-			setTimeout( function(){
-				           				
-				execute_function_1_when_function_2_returns_true( first_function_parementer, second_function_parementer, check_frequency_parameter );
-				
-			}, check_frequency_parameter );
-			
-		}
-
-	}
-	
-</script>
+function execute_function_1_when_function_2_returns_true (functionOne, functionTwo) {
+  if (functionTwo) {
+    functionOne();
+  }
+  else {
+    setTimeout(function () {
+      execute_function_1_when_function_2_returns_true(functionOne, functionTwo);
+    }, 100);
```

Comment:
I'm not sure that `check_frequency_parameter` is never defined, there's certainly a chance it could be.
I'd recommend bringing back the parameter, but changing it to get its default value to be 100 if the number you'd get from parsing the parameter doesn't make sense for a setTimeout...

```js
setTimeout(function() {
  execute_function_1_when_function_2_returns_true(first_function, second_function);
}, (check_frequency > 0 ? check_frequency : 100));
```